m2cgen/interpreters/interpreter.py

```diff
@@ -190,13 +201,10 @@ def interpret_tanh_expr(self, expr, **kwargs):
         nested_result = self._do_interpret(expr.expr, **kwargs)
         return self._cg.function_invocation(self.tanh_function_name, nested_result)
 
-    def interpret_pow_expr(self, expr, **kwargs):
-        if self.power_function_name is NotImplemented:
-            raise NotImplementedError("Power function is not provided")
-        self.with_math_module = True
-        base_result = self._do_interpret(expr.base_expr, **kwargs)
-        exp_result = self._do_interpret(expr.exp_expr, **kwargs)
-        return self._cg.function_invocation(self.power_function_name, base_result, exp_result)
+    def _wrap_infix_expr(self, expr, left_precedence, right_precedence):
+        wrap = left_precedence is not None and left_precedence > expr.precedence
+        wrap |= right_precedence is not None and right_precedence >= expr.precedence
```

Comment:
Maybe we can avoid `is not None` by changing default precedence (see comment above)?